pymc3/backends/base.py

```diff
@@ -313,6 +313,9 @@ def chains(self):
     def report(self):
         return self._report
 
+    def __iter__(self):
```

Comment:
Can you add a minimal test to make sure it raises the exception if you try to iterate over a trace?